glean-core/android/src/main/java/mozilla/telemetry/glean/private/JweMetricType.kt

```diff
@@ -0,0 +1,172 @@
+     *      Can be an empty octet sequence.
+     */
+    fun set(header: String, key: String, initVector: String, cipherText: String, authTag: String) {
+        if (disabled) {
+            return
+        }
+
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.launch {
+            LibGleanFFI.INSTANCE.glean_jwe_set(this@JweMetricType.handle, header, key, initVector, cipherText, authTag)
+        }
+    }
+
+    /**
+     * Tests whether a value is stored for the metric for testing purposes only. This function will
+     * attempt to await the last task (if any) writing to the the metric's storage engine before
+     * returning a value.
+     *
+     * @param pingName represents the name of the ping to retrieve the metric for.
+     *                 Defaults to the first value in `sendInPings`.
+     * @return true if metric value exists, otherwise false
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    @JvmOverloads
+    fun testHasValue(pingName: String = sendInPings.first()): Boolean {
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.assertInTestingMode()
+
+        val res = LibGleanFFI.INSTANCE.glean_jwe_test_has_value(this.handle, pingName)
+        return res.toBoolean()
+    }
+
+    /**
+     * Returns the stored value for testing purposes only. This function will attempt to await the
+     * last task (if any) writing to the the metric's storage engine before returning a value.
+     *
+     * @param pingName represents the name of the ping to retrieve the metric for.
+     *                 Defaults to the first value in `sendInPings`.
+     * @return value of the stored metric
+     * @throws [NullPointerException] if no value is stored
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    @JvmOverloads
+    fun testGetValue(pingName: String = sendInPings.first()): String {
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.assertInTestingMode()
+
+        if (!testHasValue(pingName)) {
+            throw NullPointerException()
+        }
+        val ptr = LibGleanFFI.INSTANCE.glean_jwe_test_get_value_as_json_string(this.handle, pingName)!!
+        return ptr.getAndConsumeRustString()
+    }
+
+    /**
+     * Returns the stored value in the compact representation for testing purposes only.
+     * This function will attempt to await the last task (if any)
+     * writing to the the metric's storage engine before returning a value.
+     *
+     * @param pingName represents the name of the ping to retrieve the metric for.
+     *                 Defaults to the first value in `sendInPings`.
+     * @return value of the stored metric
+     * @throws [NullPointerException] if no value is stored
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    @JvmOverloads
+    fun testGetValueAsPeriodDelimitedString(pingName: String = sendInPings.first()): String {
```

Comment:
```suggestion
    fun testGetCompactRepresentation(pingName: String = sendInPings.first()): String {
```

Maybe that's clearer?

Reply:
Maybe. The name came from the [the document](https://docs.google.com/document/d/1nntNIiE6braTGzoKf-lx21OVDd8ssyeIeJu3jnQQfEE/edit?usp=sharing).

Reply:
I don't have strong opinions though, if yo uthink it's best I can change.

Reply:
Yeah, I know the source (me :-) ) - I just wonder if this makes sense, since we already call it compact representation above. Up to you to decide!

Reply:
Sure, let's make it consistent. I'll change.